core/commonMain/src/implementations/immutableMap/TrieNode.kt

```diff
@@ -436,6 +439,88 @@ internal class TrieNode<K, V>(
         return this
     }
 
+    private fun mutableCollisionPutAll(otherNode: TrieNode<K, V>,
+                                       intersectionCounter: DeltaCounter,
+                                       owner: MutabilityOwnership): TrieNode<K, V> {
+        assert(nodeMap == 0)
+        assert(dataMap == 0)
+        assert(otherNode.nodeMap == 0)
+        assert(otherNode.dataMap == 0)
+        val tempBuffer = this.buffer.copyOf(newSize = this.buffer.size + otherNode.buffer.size)
+        var j = 0
+        for (i in this.buffer.size until tempBuffer.size step ENTRY_SIZE) {
+            @Suppress("UNCHECKED_CAST")
+            if (!this.collisionContainsKey(otherNode.buffer[j] as K)) {
+                tempBuffer[i] = otherNode.buffer[j]
```

Comment:
Seems like the indices got mixed up.
This should probably have looked like `tempBuffer[j] = otherNode.buffer[i - this.buffer.size]`.
Another option is to iterate through `otherNode.buffer`:
```
         var j = 0
         val otherBuffer = otherNode.buffer
         for (i in 0 until otherBuffer.size step ENTRY_SIZE) {
             @Suppress("UNCHECKED_CAST")
             if (!this.collisionContainsKey(otherBuffer[i] as K)) {
                 tempBuffer[j] = otherBuffer[i]
                 tempBuffer[j + 1] = otherBuffer[i + 1]
                 j += ENTRY_SIZE
             }
             ...
```